Source/Core/DiscIO/NANDImporter.cpp

```diff
@@ -188,40 +190,69 @@ void NANDImporter::ProcessFile(const NANDFSTEntry& entry, const std::string& par
   }
 }
 
-void NANDImporter::ExtractCertificates(const std::string& nand_root)
+bool NANDImporter::ExtractCertificates(const std::string& nand_root)
 {
-  const std::string title_contents_path =
-      nand_root + "/title/00000001/0000000d/content/00000011.app";
-  File::IOFile file(title_contents_path, "rb");
-  if (!file)
+  const std::string content_dir = nand_root + "/title/00000001/0000000d/content/";
+
+  File::IOFile tmd_file(content_dir + "title.tmd", "rb");
+  std::vector<u8> tmd_bytes(tmd_file.GetSize());
+  if (!tmd_file.ReadBytes(tmd_bytes.data(), tmd_bytes.size()))
   {
-    PanicAlertT("Unable to open %s! Refer to "
-                "https://dolphin-emu.org/docs/guides/wii-network-guide/ to set up "
-                "certificates.",
-                title_contents_path.c_str());
-    return;
+    ERROR_LOG(DISCIO, "ExtractCertificates: Could not read IOS13 TMD");
+    return false;
   }
 
-  struct Certificate
+  IOS::ES::TMDReader tmd(std::move(tmd_bytes));
+  IOS::ES::Content content_metadata;
+  tmd.GetContent(1, &content_metadata);
+
+  File::IOFile content_file(content_dir + StringFromFormat("%08x.app", content_metadata.id), "rb");
+  std::vector<u8> content_bytes(content_file.GetSize());
+  if (!content_file.ReadBytes(content_bytes.data(), content_bytes.size()))
+  {
+    ERROR_LOG(DISCIO, "ExtractCertificates: Could not read IOS13 contents");
+    return false;
+  }
+
+  struct PEMCertificate
   {
-    u32 offset;
-    u32 size;
     std::string filename;
+    std::array<u8, 4> search_bytes;
   };
-  std::array<Certificate, 3> certificates = {{{0x92834, 1005, "/clientca.pem"},
-                                              {0x92d38, 609, "/clientcakey.pem"},
-                                              {0x92440, 897, "/rootca.pem"}}};
-  for (const Certificate& cert : certificates)
+
+  std::array<PEMCertificate, 3> certificates = {{
+      {"/clientca.pem", {{0x30, 0x82, 0x03, 0xE9}}},
+      {"/clientcakey.pem", {{0x30, 0x82, 0x02, 0x5D}}},
+      {"/rootca.pem", {{0x30, 0x82, 0x03, 0x7D}}},
+  }};
+
+  for (const PEMCertificate& certificate : certificates)
   {
-    file.Seek(cert.offset, SEEK_SET);
-    std::vector<u8> pem_cert(cert.size);
-    file.ReadBytes(pem_cert.data(), pem_cert.size());
+    const auto search_result =
+        std::search(content_bytes.begin(), content_bytes.end(), certificate.search_bytes.begin(),
+                    certificate.search_bytes.end());
+
+    if (search_result == content_bytes.end())
+    {
+      ERROR_LOG(DISCIO, "ExtractCertificates: Could not find offset for certficate '%s'",
+                certificate.filename.c_str());
+      return false;
+    }
+
+    const std::string pem_file_path = nand_root + certificate.filename;
+    const ptrdiff_t certificate_offset = std::distance(content_bytes.begin(), search_result);
+    const u16 certificate_size = (*(search_result - 2) << 8) | (*(search_result - 1));
```

Comment:
Somebody would need to tell me how to use `Common::FromBigEndian` then, because all of the variations that I've tried (`FromBigEndian<u8>`, `<u16>`, `<u8*>`, `<u16*>` and `(search_results - 2)`, `(&*(search_results - 2))`, `(&content_bytes[certificate_offset - 2])`) didn't work.